src/test/java/com/google/cloud/spanner/pgadapter/ITJdbcTest.java

```diff
@@ -0,0 +1,146 @@
+          database.getId().getDatabase(),
+          Arrays.asList(
+              "create table numbers (num int not null primary key, name varchar(100))",
+              "create table all_types ("
+                  + "col_bigint bigint not null primary key, "
+                  + "col_bool bool, "
+                  + "col_bytea bytea, "
+                  + "col_float8 float8, "
+                  + "col_int int, "
+                  + "col_numeric numeric, "
+                  + "col_timestamptz timestamptz, "
+                  + "col_varchar varchar(100))"));
+    }
+    String credentials = testEnv.getCredentials();
+    ImmutableList.Builder<String> argsListBuilder =
+        ImmutableList.<String>builder()
+            .add(
+                "-p",
+                testEnv.getProjectId(),
+                "-i",
+                testEnv.getInstanceId(),
+                "-d",
+                database.getId().getDatabase(),
+                "-s",
+                String.valueOf(testEnv.getPort()),
+                "-e",
+                testEnv.getUrl().getHost());
+    if (credentials != null) {
+      argsListBuilder.add("-c", testEnv.getCredentials());
+    }
+    String[] args = argsListBuilder.build().toArray(new String[0]);
+    server = new ProxyServer(new OptionsMetadata(args));
+    server.startServer();
+  }
+
+  @AfterClass
+  public static void teardown() {
+    if (server != null) {
+      server.stopServer();
+    }
+    testEnv.cleanUp();
+  }
+
+  @Before
+  public void insertTestData() {
+    String databaseId = database.getId().getDatabase();
+    testEnv.write(databaseId, Collections.singleton(Mutation.delete("numbers", KeySet.all())));
+    testEnv.write(databaseId, Collections.singleton(Mutation.delete("all_types", KeySet.all())));
+    testEnv.write(
+        databaseId,
+        Arrays.asList(
+            Mutation.newInsertBuilder("numbers").set("num").to(1L).set("name").to("One").build(),
+            Mutation.newInsertBuilder("all_types")
+                .set("col_bigint")
+                .to(1L)
+                .set("col_bool")
+                .to(true)
+                .set("col_bytea")
+                .to(ByteArray.copyFrom("test"))
+                .set("col_float8")
+                .to(3.14d)
+                .set("col_int")
+                .to(1)
+                .set("col_numeric")
+                .to(new BigDecimal("3.14"))
+                .set("col_timestamptz")
+                .to(Timestamp.parseTimestamp("2022-01-27T17:51:30+01:00"))
+                .set("col_varchar")
+                .to("test")
+                .build()));
+  }
+
+  @Test
+  public void testSelectHelloWorld() throws SQLException {
+    try (Connection connection =
+        DriverManager.getConnection(
+            String.format("jdbc:postgresql://localhost:%d/", testEnv.getPort()))) {
+      try (ResultSet resultSet =
+          connection.createStatement().executeQuery("SELECT 'Hello World!'")) {
```

Comment:
Yes, the wireprotocol is the same for pg-jdbc, so yes it will in theory work with other queries. However:
1. The pg-jdbc driver uses binary format for many query parameter types, so we need to support that for parameterized queries to work. That is included in #27 
2. The pg-jdbc driver sends `timestamp` and `date` parameter values as untyped parameters. Support for that is also included in #27 (only for timestamp at this moment, as we currently do not support date).
3. The pg-jdbc driver prefers the extended query mode over simple query mode (which is normally used in psql). That will probably also need some tweaking.